library/HTMLPurifier/HTMLModule/HTML5/SafeExternalStylesheets.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * https://html.spec.whatwg.org/dev/semantics.html#the-link-element
+ */
+class HTMLPurifier_HTMLModule_HTML5_SafeExternalStylesheets extends HTMLPurifier_HTMLModule
+{
+    /**
+     * @type string
+     */
+    public $name = 'HTML5_SafeExternalStylesheets';
+
+    /**
+     * @param HTMLPurifier_Config $config
+     */
+    public function setup($config)
+    {
+        $allowed = $config->get('HTML.SafeExternalStylesheet');
+
+        // https://html.spec.whatwg.org/dev/semantics.html#the-link-element
+        $this->addElement('link', 'Flow', 'Empty', null, array(
+            'rel*'  => new HTMLPurifier_AttrDef_Enum(array('stylesheet')),
+            'type'  => new HTMLPurifier_AttrDef_Enum(array('text/css')),
+            'href*' => new HTMLPurifier_AttrDef_Enum($allowed, true),
```

Comment:
> I believe we would need additional post-transform logic ensuring that the spec requirements are met (presence of `href` or `rel` values depending on type).

Use of `*` covers this, see http://htmlpurifier.org/docs/enduser-customize.html specifically:
> Notice that we added an asterisk to the end of the action attribute to indicate that it is required.

It gets a bit more complex if you follow the [spec](https://html.spec.whatwg.org/dev/semantics.html#the-link-element) exactly i.e. it says:
> A link element must have either a rel attribute or an itemprop attribute, but not both.

> One or both of the href or imagesrcset attributes must be present.

So use of `*` probably wouldn't work if that logic were to be implemented.